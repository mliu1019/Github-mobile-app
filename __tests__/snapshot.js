import React from "react";
import renderer from "react-test-renderer";

import Home from "../src/views/home";
import Profile from "../src/views/profile";
import Repos from "../src/views/repo";
import Follow from "../src/views/follow";
import Header from "../src/views/shared/header";

test("renders HOME correctly", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Header correctly", () => {
  const tree = renderer.create(<Header login={fakeLogin} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Profile correctly", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const tree = renderer
    .create(
      <Profile
        route={{
          params: {
            login: fakeLogin,
          },
        }}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Repos correctly", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const tree = renderer
    .create(
      <Repos
        route={{
          params: {
            login: fakeLogin,
          },
        }}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Followers correctly", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const tree = renderer
    .create(
      <Follow
        route={{
          params: {
            login: fakeLogin,
            type: "follower",
          },
        }}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders Following correctly", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const tree = renderer
    .create(
      <Follow
        route={{
          params: {
            login: fakeLogin,
            type: "following",
          },
        }}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const fakeLogin = "MirandaLiu1019";
