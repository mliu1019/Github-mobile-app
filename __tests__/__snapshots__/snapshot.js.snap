// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HOME correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "35%",
      },
    ]
  }
>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 32,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    Profile
  </Text>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 32,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    Repos
  </Text>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 32,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    Followers
  </Text>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 32,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    Following
  </Text>
</View>
`;

exports[`renders Header correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "papayawhip",
      "flexDirection": "row",
      "height": 100,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.8,
        "justifyContent": "center",
        "marginLeft": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      MirandaLiu1019
    </Text>
  </View>
</View>
`;

exports[`renders Profile correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
/>
`;
